harvester_e2e_tests/integration/test_hosts.py

```diff
@@ -67,3 +127,156 @@ def test_host_poweron_state(api_client, host_state, wait_timeout):
 
     _, node = api_client.hosts.get(node['id'])
     assert "active" == node["metadata"]["state"]["name"]
+
+
+@pytest.mark.hosts
+@pytest.mark.p0
+def test_verify_host_info(api_client):
+    status_code, nodes_info = api_client.hosts.get()
+
+    assert status_code == 200, f"Failed to list nodes with error: {nodes_info}"
+    assert len(nodes_info['data']) >= 1, "Incorrect hosts amount"
+
+    for node in nodes_info.get('data', []):
+        node_name = node.get('metadata', {}).get('name', "")
+        assert "" != node_name, ("Node name can't be empty")
+
+        cpus = node.get('status', {}).get('capacity', {}).get('cpu', 0)
+        assert 0 != cpus, (
+            f"Node {node_name} cpu should not be zero"
+        )
+
+        mems = node.get('status', {}).get('capacity', {}).get('memory', "0Ki")
+        mems = "".join(c for c in mems if c.isdigit())
+        mems = mems or "0"
+        assert 0 != int(mems, 10), (
+            f"Node {node_name} memory should not be zero"
+        )
+
+
+@pytest.mark.hosts
+@pytest.mark.p0
+class TestHostTriggerVmMigrate:
+    @pytest.mark.dependency()
+    def test_prerequisite(self, api_client):
+        status_code, nodes_info = api_client.hosts.get()
+
+        assert status_code == 200, f"Failed to list nodes with error: {nodes_info}"
+        assert len(nodes_info['data']) > 1, "Please use multiple hosts to test vm migration"
+
+        for node_info in nodes_info.get('data', []):
+            for condition in node_info.get('status', {}).get('conditions', []):
```

Comment:
prerequisite is not a independent test item, so we would not make it as a test. (you can let it be a fixture)

for test trigger VM migration, we will just need `2` nodes to be ready, so we don't need to check all nodes.
I would suggest to change this method into fixture and return a list contains name of ready nodes.